PetAdoptions/petsite/petsite/Controllers/PetFoodController.cs

```diff
@@ -43,20 +53,40 @@ public async Task<string> Index()
         [HttpGet("/petfood-metric/{entityId}/{value}")]
         public async Task<string> PetFoodMetric(string entityId, float value)
         {
-            // X-Ray FTW
-            AWSXRayRecorder.Instance.BeginSubsegment("Calling PetFood metric");
-            Console.WriteLine("Calling: " + "http://petfood-metric/metric/" + entityId + "/" + value.ToString());
-            Console.WriteLine($"[{AWSXRayRecorder.Instance.GetEntity().TraceId}][{AWSXRayRecorder.Instance.TraceContext.GetEntity().RootSegment.TraceId}] - Calling PetFood metric");            
+            // Add custom span attributes using Activity API
+            var currentActivity = Activity.Current;
+            if (currentActivity != null)
+            {
+                currentActivity.SetTag("operation", "PetFoodMetric");
+                currentActivity.SetTag("entityId", entityId);
+                currentActivity.SetTag("value", value.ToString());
+                
+                Console.WriteLine("Calling: " + "http://petfood-metric/metric/" + entityId + "/" + value.ToString());
+            }
             
-            var httpClient = new HttpClient(new HttpClientXRayTracingHandler(new HttpClientHandler()));
-            string result = await httpClient.GetStringAsync("http://petfood-metric/metric/" + entityId + "/" + value.ToString());
-
-            AWSXRayRecorder.Instance.AddAnnotation("entityId", entityId);
-            AWSXRayRecorder.Instance.AddAnnotation("value", value.ToString());
-            AWSXRayRecorder.Instance.EndSubsegment();
+            string result;
+            
+            try
+            {
+                // Create a new activity for the API call
+                using (var activity = new Activity("Calling PetFood metric").Start())
+                {
+                    if (activity != null)
+                    {
+                        activity.SetTag("entityId", entityId);
+                        activity.SetTag("value", value.ToString());
+                    }
+                    
+                    result = await httpClient.GetStringAsync("http://petfood-metric/metric/" + entityId + "/" + value.ToString());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Potential SSRF (Server-Side Request Forgery) vulnerability detected. This pattern identifies code structures that might allow attackers to manipulate server-side requests, posing a risk of unauthorized access and data leaks. Ensure that user-controlled inputs in HttpClient, WebClient, or RestClient requests are thoroughly validated and sanitized to mitigate this security threat.